enhancements/sig-architecture/70-spread-policy/README.md

```diff
@@ -0,0 +1,226 @@
+Instead, we want to place more workload on us-east-1 than us-west-1.
+Besides, we want to control such preferences by user-specified configurations.
+
+**Example.** We want six clusters to deploy our backend service.
+Moreover, we want 2/3 of them to be located in us-east-1 and 1/3 to be located in us-west-1 since there are more clients from the east coast of the US.
+
+## Goal
+Generally, we have two goals for the spread policy:
+1. Cover the full functionality in the Motivation section;
+2. Implement a high-performance and scalable mechanism that works for thousands of clusters efficiently.
+
+## Proposal
+
+### API Spec
+```go
+type PlacementSpec struct {
+	// ...
+
+	// SpreadPolicy defines how placement decision should be distributed among a
+	// set of clusters.
+	SpreadPolicy []SpreadPolicyTerm `json:"spreadPolicy,omitempty"`
+}
+
+// SpreadPolicyTerm defines a terminology to spread placement decisions
+type SpreadPolicyTerm struct {
+	// TopologyKey is either a label key or a cluster claim name of ManagedClusters
+	// +required
+	TopologyKey string `json:"topologyKey"`
+
+	// TopologyKeyType indicates the type of TopologyKey. It could be Label or Claim.
+	// +required
+	TopologyKeyType string `json:"topologyKeyType"`
+
+	// MaxSkew describes the degree to which the workload may be unevenly distributed.
+	// If omitted, the scheduler treat this term a soft constraint. Otherwise, the
+	// scheduler would not schedule if MaxSkew cannot be satisfied (i.e., a hard constraint).
+	MaxSkew *int32 `json:"maxSkew,omitempty"`
+
+	// Weight indicates the weight of the SpreadPolicyTerm. It is required only when multiple
+	// terms are used. To spread the workload, the scheduler takes the even constraints with
+	// higher weight into prior consideration.
+	Weight int32 `json:"weight,omitempty"`
+}
+```
+
+### API Example
+
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: Placement
+metadata:
+  name: demo-placement
+spec:
+  # numberOfClusters is required if spreadConstraints contains
+  # a constraint of Even type.
+  numberOfClusters: 4
+  # Prediates work before the spread constraints to filter
+  # out some undesired topologies.
+  predicates:
+  - # ...
+  # SpreadPolicy contain two terms, which work
+  # independently and jointly.
+  spreadPolicy:
+    # When multiple terms are used, the user should specify their weights.
+    # The scheduler will consider to first do the spread according to the
+    # terms with higher weight.
+    # In this example, the scheduler will first spread the workload
+    # by providers.
+    # Within each provider, the scheduler then spread the workload by
+    # regions. 
+  - topologyKey: provider
+    topologyKeyType: Label
+    maxSkew: 2
+    weight: 20
+    # maxSkew is omitted as this term is a soft constraint
+  - topologyKey: region
+    topologyKeyType: Label
+    weight: 10
+  # I plan to exclude the spread policy in Additive mode.
```

Comment:
what is the reason for this?

Reply:
Hmmm, in my opinion it is rather a advanced feature. If we include it into Additive mode, we need to provide a default weight, which may be hard to set.

Reply:
By default, the plugin could be disabled, The reason of using additive is if we do not need to add other plugin (balance or steady) when setting this plugin